--- ./src/http/ngx_http_core_module.c.orig	2015-09-25 12:47:30.325494449 +0300
+++ ./src/http/ngx_http_core_module.c	2015-09-25 12:47:03.229494964 +0300
@@ -146,6 +146,7 @@
 static ngx_conf_enum_t  ngx_http_gzip_http_version[] = {
     { ngx_string("1.0"), NGX_HTTP_VERSION_10 },
     { ngx_string("1.1"), NGX_HTTP_VERSION_11 },
+    { ngx_string("2.0"), NGX_HTTP_VERSION_20 },
     { ngx_null_string, 0 }
 };
 

