--- src/ngx_http_lua_timer.c	2015-05-26 19:38:00.020075434 +0300
+++ src/ngx_http_lua_timer.c_orig	2015-05-26 19:37:00.664075441 +0300
@@ -350,7 +350,7 @@
 
 #if defined(nginx_version) && nginx_version >= 1003014
 
-    ngx_http_set_connection_log(r->connection, clcf->error_log);
+    ngx_set_connection_log(r->connection, clcf->error_log);
 
 #else

--- src/ngx_http_lua_initworkerby.c	2015-05-26 19:37:34.132075437 +0300
+++ src/ngx_http_lua_initworkerby.c_orig	2015-05-26 19:36:52.508075442 +0300
@@ -227,7 +227,7 @@
 
 #if defined(nginx_version) && nginx_version >= 1003014
 
-    ngx_http_set_connection_log(r->connection, clcf->error_log);
+    ngx_set_connection_log(r->connection, clcf->error_log);
 
 #else
 

