diff --git a/lua/ngx/ssl.lua b/lua/ngx/ssl.lua
index 9f90998..4bbd0b6 100644
--- a/lua/ngx/ssl.lua
+++ b/lua/ngx/ssl.lua
@@ -39,6 +39,9 @@ int ngx_http_lua_ffi_ssl_server_name(ngx_http_request_t *r, char **name,
 
 int ngx_http_lua_ffi_cert_pem_to_der(const unsigned char *pem, size_t pem_len,
     unsigned char *der, char **err);
+    
+int ngx_http_lua_ffi_privkey_pem_to_der(const unsigned char *pem, size_t pem_len,
+    unsigned char *der, char **err);
 
 int ngx_http_lua_ffi_ssl_get_ocsp_responder_from_der_chain(
     const char *chain_data, size_t chain_len, char *out, size_t *out_size,
@@ -162,6 +165,23 @@ function _M.server_name()
 end
 
 
+function _M.privkey_pem_to_der(pem)
+    local r = getfenv(0).__ngx_req
+    if not r then
+        return error("no request found")
+    end
+
+    local outbuf = get_string_buf(#pem)
+
+    local sz = C.ngx_http_lua_ffi_privkey_pem_to_der(pem, #pem, outbuf, errmsg)
+    if sz > 0 then
+        return ffi_str(outbuf, sz)
+    end
+
+    return nil, ffi_str(errmsg[0])
+end
+
+
 function _M.cert_pem_to_der(pem)
     local r = getfenv(0).__ngx_req
     if not r then
diff --git a/src/ngx_http_lua_sslcertby.c b/src/ngx_http_lua_sslcertby.c
index 97f6311..c7ef482 100644
--- a/src/ngx_http_lua_sslcertby.c
+++ b/src/ngx_http_lua_sslcertby.c
@@ -719,6 +719,38 @@ ngx_http_lua_ffi_ssl_server_name(ngx_http_request_t *r, char **name,
     return NGX_DECLINED;
 }
 
+int
+ngx_http_lua_ffi_privkey_pem_to_der(const u_char *pem, size_t pem_len, u_char *der,
+    char **err)
+{
+    int       total;
+    BIO      *bio;
+    EVP_PKEY *pkey;
+
+    bio = BIO_new_mem_buf((char *) pem, (int) pem_len);
+    if (bio == NULL) {
+        *err = "BIO_new_mem_buf() failed";
+        return NGX_ERROR;
+    }
+
+    pkey = PEM_read_bio_PrivateKey(bio, NULL, NULL, NULL);
+    if (pkey == NULL) {
+        *err = "PEM_read_bio_PrivateKey() failed";
+        BIO_free(bio);
+        return NGX_ERROR;
+    }
+
+    total = i2d_PrivateKey(pkey, &der);
+    if (total < 0) {
+        EVP_PKEY_free(pkey);
+        BIO_free(bio);
+        return NGX_ERROR;
+    }
+
+    EVP_PKEY_free(pkey);
+    BIO_free(bio);
+    return total;
+}
 
 int
 ngx_http_lua_ffi_cert_pem_to_der(const u_char *pem, size_t pem_len, u_char *der,
