Index: src/main/java/uk/me/phillsacre/uploader/util/PropertyUtils.java
===================================================================
--- src/main/java/uk/me/phillsacre/uploader/util/PropertyUtils.java	(revision 227)
+++ src/main/java/uk/me/phillsacre/uploader/util/PropertyUtils.java	(revision 228)
@@ -101,22 +101,20 @@
 				// Copy across properties file to userhome to user will use it next time
 				File propsDir = new File(System.getProperty("user.home")
 						+ File.separator + ".fb-uploader");
+
 				if (!propsDir.isDirectory())
 				{
 					if (!propsDir.mkdir())
 					{
-						_log
-								.warn("Could not create properties directory in user home!");
+						_log.warn("Could not create .fb-uploader directory");
+						return f;
 					}
-					else
-					{
-						File newPropsFile = new File(propsDir,
-								"user.properties");
-						copyFile(f, newPropsFile);
-
-						f = newPropsFile;
-					}
 				}
+
+				File newPropsFile = new File(propsDir, "user.properties");
+				copyFile(f, newPropsFile);
+
+				f = newPropsFile;
 			}
 		}
 
