diff -Nur fb-photo-uploader-1.3.3/build.xml fb-photo-uploader-1.3.3_patched/build.xml
--- fb-photo-uploader-1.3.3/build.xml	2008-05-28 15:53:20.000000000 +0300
+++ fb-photo-uploader-1.3.3_patched/build.xml	2008-05-29 22:50:55.000000000 +0300
@@ -94,10 +94,10 @@
 		</cobertura-instrument>
 	</target>
 
-	<target name="test" depends="compile, test-init, instrument" description="Test the project">
+	<target name="test" depends="test-init" description="Test the project">
 		<javac srcdir="${test.src.dir}" destdir="${test.build.dir}" classpathref="test.classpath" debug="true" />
 
-		<junit printsummary="on" fork="true" forkmode="once">
+		<junit printsummary="on" fork="true" forkmode="once" haltonfailure="true">
 			<formatter type="xml" />
 			<sysproperty key="base.dir" value="${base.dir}" />
 			<sysproperty key="net.sourceforge.cobertura.datafile" file="${build.instrumented.dir}/cobertura.ser" />
@@ -113,11 +113,9 @@
 			<report format="frames" todir="${test.junit.report.dir}" />
 			<fileset dir="${test.data.dir}" />
 		</junitreport>
-
-		<cobertura-report format="html" destdir="${test.coverage.report.dir}" srcdir="${src.dir}" datafile="${build.instrumented.dir}/cobertura.ser" />
 	</target>
 
-	<target name="package" depends="compile, test" description="Packages the project into a JAR file">
+	<target name="package" depends="compile" description="Packages the project into a JAR file">
 		<!-- user.properties and static.properties should not be in .jar -->
 		<delete>
 			<fileset dir="${build.classes.dir}">
@@ -167,4 +165,4 @@
 	<!-- Cobertura Taskdef -->
 	<taskdef classpathref="classpath" resource="tasks.properties" />
 
-</project>
\ Dosya sonunda yenisatır yok.
+</project>
diff -Nur fb-photo-uploader-1.3.3/src/main/java/uk/me/phillsacre/uploader/gui/controllers/MainWindowController.java fb-photo-uploader-1.3.3_patched/src/main/java/uk/me/phillsacre/uploader/gui/controllers/MainWindowController.java
--- fb-photo-uploader-1.3.3/src/main/java/uk/me/phillsacre/uploader/gui/controllers/MainWindowController.java	2008-05-27 14:16:28.000000000 +0300
+++ fb-photo-uploader-1.3.3_patched/src/main/java/uk/me/phillsacre/uploader/gui/controllers/MainWindowController.java	2008-05-29 21:55:32.000000000 +0300
@@ -18,7 +18,6 @@
 import uk.me.phillsacre.uploader.util.Constants;
 import uk.me.phillsacre.uploader.util.MessageSource;
 import uk.me.phillsacre.uploader.util.PropertyUtils;
-import uk.me.phillsacre.uploader.util.VersionChecker;
 import uk.me.phillsacre.uploader.vo.AlbumVO;
 
 /**
@@ -56,23 +55,7 @@
 	 */
 	public void checkForUpdates()
 	{
-		if (_props
-				.getProperty(Constants.Properties.ENABLE_UPDATE_CHECK, "true")
-				.equalsIgnoreCase("false"))
-		{
-			_log.debug("Update check disabled -- skipping");
-			return;
-		}
-
-		VersionChecker checker = new VersionChecker();
-		Map<String, Object> newVersion = checker.getNewVersionInfo();
-
-		if (null != newVersion)
-		{
-			Date releaseDate = (Date) newVersion
-					.get(VersionChecker.RELEASE_DATE);
-			_ui.showNewReleasePopup(releaseDate);
-		}
+		//Dummy method for Gentoo
 	}
 
 	/**
diff -Nur fb-photo-uploader-1.3.3/src/main/java/uk/me/phillsacre/uploader/gui/PreferencesDialog.java fb-photo-uploader-1.3.3_patched/src/main/java/uk/me/phillsacre/uploader/gui/PreferencesDialog.java
--- fb-photo-uploader-1.3.3/src/main/java/uk/me/phillsacre/uploader/gui/PreferencesDialog.java	2008-05-28 18:37:05.000000000 +0300
+++ fb-photo-uploader-1.3.3_patched/src/main/java/uk/me/phillsacre/uploader/gui/PreferencesDialog.java	2008-05-29 22:01:14.000000000 +0300
@@ -115,14 +115,12 @@
                         .addComponent(_lookAndFeelList, 0, 268, Short.MAX_VALUE))
                     .addComponent(_approvePhotos, javax.swing.GroupLayout.DEFAULT_SIZE, 364, Short.MAX_VALUE)
                     .addComponent(_deletePhotos)
-                    .addComponent(_checkForUpdates, javax.swing.GroupLayout.Alignment.TRAILING, javax.swing.GroupLayout.DEFAULT_SIZE, 364, Short.MAX_VALUE)
                     .addComponent(_autoResize, javax.swing.GroupLayout.DEFAULT_SIZE, 364, Short.MAX_VALUE))
                 .addContainerGap())
         );
         jPanel2Layout.setVerticalGroup(
             jPanel2Layout.createParallelGroup(javax.swing.GroupLayout.Alignment.LEADING)
             .addGroup(jPanel2Layout.createSequentialGroup()
-                .addComponent(_checkForUpdates)
                 .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
                 .addComponent(_autoResize)
                 .addPreferredGap(javax.swing.LayoutStyle.ComponentPlacement.RELATED)
diff -Nur fb-photo-uploader-1.3.3/src/main/java/uk/me/phillsacre/uploader/util/Constants.java fb-photo-uploader-1.3.3_patched/src/main/java/uk/me/phillsacre/uploader/util/Constants.java
--- fb-photo-uploader-1.3.3/src/main/java/uk/me/phillsacre/uploader/util/Constants.java	2008-05-27 14:16:28.000000000 +0300
+++ fb-photo-uploader-1.3.3_patched/src/main/java/uk/me/phillsacre/uploader/util/Constants.java	2008-05-30 19:34:22.000000000 +0300
@@ -44,7 +44,7 @@
 		/**
 		 * Whether to copy properties across to the user.home directory
 		 */
-		public static final boolean COPY_PROPS_TO_USERHOME = false;
+		public static final boolean COPY_PROPS_TO_USERHOME = true;
 	}
 
 	/**
diff -Nur fb-photo-uploader-1.3.3/src/test/java/uk/me/phillsacre/uploader/gui/controllers/MainWindowControllerTest.java fb-photo-uploader-1.3.3_patched/src/test/java/uk/me/phillsacre/uploader/gui/controllers/MainWindowControllerTest.java
--- fb-photo-uploader-1.3.3/src/test/java/uk/me/phillsacre/uploader/gui/controllers/MainWindowControllerTest.java	2008-05-27 14:16:28.000000000 +0300
+++ fb-photo-uploader-1.3.3_patched/src/test/java/uk/me/phillsacre/uploader/gui/controllers/MainWindowControllerTest.java	2008-05-29 22:48:13.000000000 +0300
@@ -34,33 +34,6 @@
 				_photoListModel);
 	}
 
-	public void testCheckForUpdatesDisabled()
-	{
-		PropertyUtils props = new PropertyUtils();
-		props.setProperty(Constants.Properties.ENABLE_UPDATE_CHECK, "false");
-
-		_controller.checkForUpdates();
-
-		if (_ui.newReleasePopup)
-		{
-			fail("Check for updates is disabled but user has been shown new release popup");
-		}
-	}
-
-	public void testCheckForUpdatesEnabled()
-	{
-		PropertyUtils props = new PropertyUtils();
-		props.setProperty(Constants.Properties.ENABLE_UPDATE_CHECK, "true");
-		props.setProperty(Constants.Properties.BUILD_NUMBER, "0");
-
-		_controller.checkForUpdates();
-
-		if (!_ui.newReleasePopup)
-		{
-			fail("User has not been shown check for updates popup");
-		}
-	}
-
 	public void testAddPhoto()
 	{
 		Photo photo = new PhotoImpl(TestUtils.getTestImageFile());
