diff -Nur fb-photo-uploader-1.2.1/build.xml fb-photo-uploader-1.2.1_patched/build.xml
--- fb-photo-uploader-1.2.1/build.xml	2007-06-28 19:10:03.000000000 +0300
+++ fb-photo-uploader-1.2.1_patched/build.xml	2007-11-11 00:58:52.000000000 +0200
@@ -70,7 +70,7 @@
 		</copy>
 	</target>
 
-	<target name="test" depends="compile, test-init" description="Test the project">
+	<target name="test" depends="test-init" description="Test the project">
 		<javac srcdir="${test.src.dir}" destdir="${test.build.dir}" classpathref="test.classpath" debug="true" />
 
 		<junit printsummary="on">
@@ -89,7 +89,7 @@
 		</junitreport>
 	</target>
 
-	<target name="package" depends="compile, test" description="Packages the project into a JAR file">
+	<target name="package" depends="compile" description="Packages the project into a JAR file">
 		<jar destfile="${package.dir}/bin/fb-uploader.jar" basedir="${build.dir}" />
 	</target>
 
@@ -113,4 +113,4 @@
 		<zip basedir="${package.dir}" destfile="${dist.dir}/fb-uploader.zip" />
 	</target>
 
-</project>
\ Dosya sonunda yenisatır yok.
+</project>
diff -Nur fb-photo-uploader-1.2.1/src/main/java/uk/me/phillsacre/uploader/gui/MainWindow.java fb-photo-uploader-1.2.1_patched/src/main/java/uk/me/phillsacre/uploader/gui/MainWindow.java
--- fb-photo-uploader-1.2.1/src/main/java/uk/me/phillsacre/uploader/gui/MainWindow.java	2007-11-01 18:51:04.000000000 +0200
+++ fb-photo-uploader-1.2.1_patched/src/main/java/uk/me/phillsacre/uploader/gui/MainWindow.java	2007-11-11 00:53:02.000000000 +0200
@@ -101,8 +101,6 @@
 			_instance.setVisible(true);
 			_instance.initFacebook();
 			_instance.refreshAlbumList();
-
-			_instance.checkForUpdates();
 		}
 
 		return _instance;
@@ -154,16 +152,6 @@
 	 */
 	private void initComponents()
 	{
-		try
-		{
-			UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
-		}
-		catch (Exception e)
-		{
-			throw new UploaderException(MessageSource
-					.getMessage("error.couldNotSetLookAndFeel"), e);
-		}
-
 		MainWindowActionListener actionListener = new MainWindowActionListener();
 
 		getContentPane().setBackground(SystemColor.control);
@@ -426,20 +414,6 @@
 		menu.add(menuItem);
 		menu.addSeparator();
 
-		boolean enableUpdates = _props.getProperty(
-				"uploader.enable-update-check", "true")
-				.equalsIgnoreCase("true");
-
-		menuItem = new JCheckBoxMenuItem(MessageSource
-				.getMessage("mainwindow.menu.enableCheckForUpdates"),
-				enableUpdates);
-		menuItem.setToolTipText(MessageSource
-				.getMessage("mainwindow.menu.enableCheckForUpdates.tooltip"));
-		menuItem.addActionListener(actionListener);
-		menu.add(menuItem);
-
-		menu.addSeparator();
-
 		menu.add(createImportSubmenu());
 
 		menu.addSeparator();
diff -Nur fb-photo-uploader-1.2.1/src/main/java/uk/me/phillsacre/uploader/util/PropertyUtils.java fb-photo-uploader-1.2.1_patched/src/main/java/uk/me/phillsacre/uploader/util/PropertyUtils.java
--- fb-photo-uploader-1.2.1/src/main/java/uk/me/phillsacre/uploader/util/PropertyUtils.java	2007-11-01 18:51:04.000000000 +0200
+++ fb-photo-uploader-1.2.1_patched/src/main/java/uk/me/phillsacre/uploader/util/PropertyUtils.java	2007-11-11 01:25:22.000000000 +0200
@@ -29,27 +29,33 @@
 
 		try
 		{
-			// First, attempt to load the properties file from the user.home
-			File f = new File(System.getProperty("user.home") + File.separator
-					+ ".fb-uploader" + File.separator + "uploader.properties");
-
-			if (f.exists() && f.isFile())
-			{
-				_log.debug("Using properties file from user home: [" + f.getAbsolutePath() + "]");
-				_configFile = f;
-			}
-			else
-			{
-				_log.debug("Using properties file from ClassPath");
-				_configFile = new File(ClassLoader.getSystemClassLoader()
-						.getResource("uploader.properties").toURI());
-			}
-
+			_configFile = new File(System.getProperty("user.home")+"/.fb-photo-uploader.properties");
 			_configuration.load(_configFile);
 		}
 		catch (Exception e)
 		{
-			throw new UploaderException("Could not load properties", e);
+			_configuration.addProperty("uploader.resize.1","800 x 600");
+			_configuration.addProperty("uploader.resize.1","1024 x 768");
+			_configuration.addProperty("uploader.resize.1","1280 x 960");
+			_configuration.addProperty("facebook.api-key","99e18912a9ad6e2333146e0bd231425f");
+			_configuration.addProperty("facebook.secret","9a50e37e7eba996fac297689251e177c");
+			_configuration.addProperty("facebook.login-url","http://www.facebook.com/login.php?v=1.0");
+			_configuration.addProperty("session.persistent-key","");
+			_configuration.addProperty("session.secret-key","");
+			_configuration.addProperty("session.user-id","");
+			_configuration.addProperty("uploader.use-mock-dao","false");
+			_configuration.addProperty("uploader.editor.use-scaling","true");
+			_configuration.addProperty("uploader.last-directory","");
+			_configuration.addProperty("app.build-number","297");
+			_configuration.addProperty("uploader.auto-resize","");
+			try
+			{
+				_configuration.save(_configFile);
+			}
+			catch (Exception ex)
+			{
+				throw new UploaderException("Could not save properties", ex);
+			}
 		}
 	}
 
