diff -Nur fb-photo-uploader-1.2.1/build.xml fb-photo-uploader-1.2.1_patched/build.xml
--- fb-photo-uploader-1.2.1/build.xml	2007-06-28 19:10:03.000000000 +0300
+++ fb-photo-uploader-1.2.1_patched/build.xml	2007-11-11 00:58:52.000000000 +0200
@@ -70,7 +70,7 @@
 		</copy>
 	</target>
 
-	<target name="test" depends="compile, test-init" description="Test the project">
+	<target name="test" depends="test-init" description="Test the project">
 		<javac srcdir="${test.src.dir}" destdir="${test.build.dir}" classpathref="test.classpath" debug="true" />
 
 		<junit printsummary="on">
@@ -89,7 +89,7 @@
 		</junitreport>
 	</target>
 
-	<target name="package" depends="compile, test" description="Packages the project into a JAR file">
+	<target name="package" depends="compile" description="Packages the project into a JAR file">
 		<jar destfile="${package.dir}/bin/fb-uploader.jar" basedir="${build.dir}" />
 	</target>
 
@@ -113,4 +113,4 @@
 		<zip basedir="${package.dir}" destfile="${dist.dir}/fb-uploader.zip" />
 	</target>
 
-</project>
\ Dosya sonunda yenisatır yok.
+</project>
diff -Nur fb-photo-uploader-1.2.1/src/main/java/uk/me/phillsacre/uploader/gui/MainWindow.java fb-photo-uploader-1.2.1_patched/src/main/java/uk/me/phillsacre/uploader/gui/MainWindow.java
--- fb-photo-uploader-1.2.1/src/main/java/uk/me/phillsacre/uploader/gui/MainWindow.java	2007-11-01 18:51:04.000000000 +0200
+++ fb-photo-uploader-1.2.1_patched/src/main/java/uk/me/phillsacre/uploader/gui/MainWindow.java	2007-11-11 00:53:02.000000000 +0200
@@ -101,8 +101,6 @@
 			_instance.setVisible(true);
 			_instance.initFacebook();
 			_instance.refreshAlbumList();
-
-			_instance.checkForUpdates();
 		}
 
 		return _instance;
@@ -154,16 +152,6 @@
 	 */
 	private void initComponents()
 	{
-		try
-		{
-			UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
-		}
-		catch (Exception e)
-		{
-			throw new UploaderException(MessageSource
-					.getMessage("error.couldNotSetLookAndFeel"), e);
-		}
-
 		MainWindowActionListener actionListener = new MainWindowActionListener();
 
 		getContentPane().setBackground(SystemColor.control);
@@ -426,20 +414,6 @@
 		menu.add(menuItem);
 		menu.addSeparator();
 
-		boolean enableUpdates = _props.getProperty(
-				"uploader.enable-update-check", "true")
-				.equalsIgnoreCase("true");
-
-		menuItem = new JCheckBoxMenuItem(MessageSource
-				.getMessage("mainwindow.menu.enableCheckForUpdates"),
-				enableUpdates);
-		menuItem.setToolTipText(MessageSource
-				.getMessage("mainwindow.menu.enableCheckForUpdates.tooltip"));
-		menuItem.addActionListener(actionListener);
-		menu.add(menuItem);
-
-		menu.addSeparator();
-
 		menu.add(createImportSubmenu());
 
 		menu.addSeparator();
