From d1fd201b4fa19b6612146a24f1e88a81054fd08a Mon Sep 17 00:00:00 2001
From: Jakub Kruszona-Zawadzki <acid@moosefs.com>
Date: Tue, 12 Nov 2019 06:52:35 +0100
Subject: [PATCH] (cli+cgi) fixed bug related to masterinfolist (closes #308)

---
 NEWS                    | 4 ++++
 mfsscripts/mfscli.py.in | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/NEWS b/NEWS
index ade44a4..e05d69e 100644
--- a/NEWS
+++ b/NEWS
@@ -1,5 +1,9 @@
 This file lists noteworthy changes in MooseFS.
 
+* MooseFS 3.0.108-1 (WIP)
+
+  - (cli+cgi) fixed bug related to masterinfolist
+
 * MooseFS 3.0.107-1 (2019-11-04)
 
   - (master) added NOP command to changelog
diff --git a/mfsscripts/mfscli.py.in b/mfsscripts/mfscli.py.in
index 0a1487e..c48762b 100644
--- a/mfsscripts/mfscli.py.in
+++ b/mfsscripts/mfscli.py.in
@@ -1957,7 +1957,7 @@ else:
 master_minusectime = None
 master_maxusectime = None
 if leader_usectime==None or leader_usectime==0:
-	for sortip,strip,sortver,strver,statestr,statecolor,metaversion,memusage,syscpu,usercpu,lastsuccessfulstore,lastsaveseconds,lastsavestatus,exportschecksum,metaid,lastsavemetaversion,lastsavemetachecksum,usectime,chlogtime in masterlistinfo:
+	for sortip,strip,sortver,strver,statestr,statecolor,metaversion,memusage,syscpu,usercpu,lastsuccessfulstore,lastsaveseconds,lastsavestatus,exportschecksum,usectime,chlogtime in masterlistinfo:
 		if usectime!=None and usectime>0:
 			if master_minusectime==None or usectime<master_minusectime:
 				master_minusectime = usectime
