--- /usr/local/tmp/portage/dev-java/maven-script-2.0.6/work/maven-script-2.0.6/maven-script-ant/src/main/java/org/apache/maven/script/ant/AntMojoComponentFactory.java	2007-05-01 19:35:42.000000000 +0200
+++ AntMojoComponentFactory.java	2007-05-02 01:08:29.511539108 +0200
@@ -30,7 +30,7 @@
     extends AntComponentFactory
 {
 
-    public Object newInstance( ComponentDescriptor descriptor, ClassRealm realm, PlexusContainer container )
+    public Object newInstance( ComponentDescriptor descriptor, org.codehaus.plexus.classworlds.realm.ClassRealm  realm, PlexusContainer container )
         throws ComponentInstantiationException
     {
         return new AntMojoWrapper( (AntScriptInvoker) super.newInstance( descriptor, realm, container ) );
