--- /usr/local/tmp/portage/dev-java/plexus-jetty-httpd-1.0/work/plexus-jetty-httpd-1.0/src/main/java/org/codehaus/plexus/jetty/DefaultHttpd.java	2007-04-21 19:44:39.000000000 +0200
+++ DefaultHttpd.java	2007-04-21 21:09:29.327604296 +0200
@@ -80,6 +80,10 @@
         {
             throw new StartingException( "Error starting the jetty httpd server: ", e );
         }
+         catch ( Exception e )
+        {
+            throw new StartingException( "Error starting the jetty httpd server: ", e );
+        } 
     }
 
     public void stop()
@@ -94,4 +98,4 @@
             throw new StoppingException( "Error stopping the jetty httpd server: ", e );
         }
     }
-}
\ Pas de fin de ligne à la fin du fichier.
+}
