--- src/main/java/org/codehaus/plexus/jetty/DefaultHttpd.java	2007-05-20 08:54:15.000000000 +1200
+++ src/main/java/org/codehaus/plexus/jetty/DefaultHttpd.java	2007-05-20 09:06:39.000000000 +1200
@@ -76,7 +76,7 @@
         {
             server.start();
         }
-        catch ( MultiException e )
+        catch ( Exception e )
         {
             throw new StartingException( "Error starting the jetty httpd server: ", e );
         }
@@ -94,4 +94,4 @@
             throw new StoppingException( "Error stopping the jetty httpd server: ", e );
         }
     }
-}
\ No newline at end of file
+}
