diff -ur plexus-jetty-httpd-1.0_beta2/src/main/java/org/codehaus/plexus/jetty/DefaultHttpd.java new/src/main/java/org/codehaus/plexus/jetty/DefaultHttpd.java
--- plexus-jetty-httpd-1.0_beta2/src/main/java/org/codehaus/plexus/jetty/DefaultHttpd.java	2007-05-01 23:31:07.000000000 +0200
+++ new/src/main/java/org/codehaus/plexus/jetty/DefaultHttpd.java	2007-05-25 19:14:04.987973224 +0200
@@ -76,7 +76,7 @@
         {
             server.start();
         }
-        catch ( MultiException e )
+        catch ( Exception e )
         {
             throw new StartingException( "Error starting the jetty httpd server: ", e );
         }
@@ -94,4 +94,4 @@
             throw new StoppingException( "Error stopping the jetty httpd server: ", e );
         }
     }
-}
\ Pas de fin de ligne à la fin du fichier.
+}
Seulement dans new/src/main/resources/META-INF: maven
