--- bin/kibana_orig	2014-11-11 22:57:34.000000000 +0300
+++ bin/kibana	2014-11-17 15:55:09.064050529 +0300
@@ -14,8 +14,6 @@
   fi
 done
 
-DIR=$(dirname "${SCRIPT}")
-
 if [ -x "${JAVA_HOME}/bin/java" ]; then
   JAVA="${JAVA_HOME}/bin/java"
 else
@@ -36,9 +34,9 @@
 
 # shellcheck disable=SC2086
 KIBANA_VERSION=4.0.0-BETA2 \
-  CONFIG_PATH=${DIR}/../config/kibana.yml \
+  CONFIG_PATH=/etc/kibana/kibana.yml \
   PLUGINS_FOLDER=${DIR}/../plugins \
   RACK_ENV=production \
   exec "${JAVA}" \
   $JAVA_OPTS \
-  -jar "${DIR}/../lib/kibana.jar" "$@"
\ No newline at end of file
+  -jar "/opt/kibana/kibana.jar" "$@"

