--- Makefile.am	2008-06-21 15:57:59.000000000 +1200
+++ Makefile.am	2008-06-21 15:59:04.000000000 +1200
@@ -40,6 +40,7 @@
 JAR = @JAR@
 JAVAC = @JAVAC@
 JAVAH = @JAVAH@
+JAVACFLAGS=@JAVACFLAGS@
 
 SOFLAGS=        @SOFLAGS@
 LIBMAJOR=       @XMLJ_VERSION_MAJOR@
@@ -651,7 +652,7 @@
 $(javax_xml_jar_CLASSES) $(org_w3c_dom_jar_CLASSES) $(org_xml_sax_jar_CLASSES) $(gnu_xml_jar_CLASSES): classes
 
 $(javax_xml_jar_CLASSES) $(org_w3c_dom_jar_CLASSES) $(org_xml_sax_jar_CLASSES) $(gnu_xml_jar_CLASSES): classes/%.class: $(srcdir)/source/%.java
-	$(JAVAC) -g -d classes -classpath @srcdir@/source $<
+	$(JAVAC) $(JAVACFLAGS) -g -d classes -classpath @srcdir@/source $<
 #	$(GCJ) $(GCJ_CLASS_FLAGS) -g -C -d classes $<
 
 classes:
