diff --git a/src/lxc/conf.c b/src/lxc/conf.c
index 8f1cba9..fe574ac 100644
--- a/src/lxc/conf.c
+++ b/src/lxc/conf.c
@@ -2290,10 +2290,8 @@ int lxc_setup(const char *name, struct lxc_conf *lxc_conf)
 		return -1;
 	}
 
-	if (setup_kmsg(&lxc_conf->rootfs, &lxc_conf->console)) {
+	if (setup_kmsg(&lxc_conf->rootfs, &lxc_conf->console))  // don't fail
 		ERROR("failed to setup kmsg for '%s'", name);
-		return -1;
-	}
 
 	if (setup_tty(&lxc_conf->rootfs, &lxc_conf->tty_info, lxc_conf->ttydir)) {
 		ERROR("failed to setup the ttys for '%s'", name);
