--- build.xml	2008-02-17 22:37:26.000000000 +0200
+++ build.xml.fixed	2008-02-17 22:38:30.000000000 +0200
@@ -194,11 +194,9 @@
                 </exec>
         </target>
 	
-	<taskdef name="launch4j" classname="net.sf.launch4j.ant.Launch4jTask"
-		classpath="lib/buildonly/launch4j.jar
-        :lib/buildonly/xstream.jar"/>
-
 	<target name="dist-win32" description="Copy the stuff for Windows to the dist dir">
+		<taskdef name="launch4j" classname="net.sf.launch4j.ant.Launch4jTask"
+		classpath="lib/buildonly/launch4j.jar:lib/buildonly/xstream.jar"/>
 		<delete dir="${dist.dir}/win32/${app.name}-win32/"/>
 		<copy todir="${dist.dir}/win32/${app.name}-win32/lib">
 			<fileset file="${lib.dir}/*.jar"/>
