--- Makefile.orig	2017-03-02 14:11:20.086338186 +0300
+++ Makefile	2017-03-02 14:12:16.303004569 +0300
@@ -157,7 +157,7 @@
 #
 # Header and Libraries of CUDA
 #
-CUDA_PATH_LIST := /usr/local/cuda /usr/local/cuda-*
+CUDA_PATH_LIST := /opt/cuda /usr/local/cuda-*
 CUDA_PATH := $(shell for x in $(CUDA_PATH_LIST);    \
            do test -e "$$x/include/cuda.h" && echo $$x; done | head -1)
 IPATH := $(CUDA_PATH)/include

