diff -Nur zekr/res/ui/theme/sky/sura.vm zekr_fixed/res/ui/theme/sky/sura.vm
--- zekr/res/ui/theme/sky/sura.vm	2007-09-11 23:52:54.000000000 +0300
+++ zekr_fixed/res/ui/theme/sky/sura.vm	2007-09-30 07:25:34.000000000 +0300
@@ -23,7 +23,7 @@
 	#set($TRAIL="right")
 #end
 #macro(putBismillah)
-	#if ($SURA_NUM != 9) ##&& $SURA != 1) #** not a good idea! *#
+	#if ($SURA_NUM != 9) ##&& $SURA != 1)
 	<img src="$BISM_PATH" alt="بسم الله الرحمن الرحیم">
 	#end
 #end
@@ -79,4 +79,4 @@
 </div>
 #parse("$THEME_DIR/watermark.vm")
 </body>
-</html>
\ Dosya sonunda yenisatır yok.
+</html>
diff -Nur zekr/res/ui/theme/uthman-taha/sura.vm zekr_fixed/res/ui/theme/uthman-taha/sura.vm
--- zekr/res/ui/theme/uthman-taha/sura.vm	2007-09-11 23:53:04.000000000 +0300
+++ zekr_fixed/res/ui/theme/uthman-taha/sura.vm	2007-09-30 07:25:44.000000000 +0300
@@ -23,7 +23,7 @@
 	#set($TRAIL="right")
 #end
 #macro(putBismillah)
-	#if ($SURA_NUM != 9) ##&& $SURA != 1) #** not a good idea! *#
+	#if ($SURA_NUM != 9) ##&& $SURA != 1)
 	<img src="$BISM_PATH" alt="بسم الله الرحمن الرحیم">
 	#end
 #end
@@ -79,4 +79,4 @@
 </div>
 #parse("$THEME_DIR/watermark.vm")
 </body>
-</html>
\ Dosya sonunda yenisatır yok.
+</html>
