--- src/main/com/tonicsystems/jarjar/util/AntJarProcessor.java.orig	2010-02-28 01:42:53.000000000 +0100
+++ src/main/com/tonicsystems/jarjar/util/AntJarProcessor.java	2010-02-28 01:43:05.000000000 +0100
@@ -84,7 +84,7 @@
         String dir = file.substring(0, slash);
         if (dirs.add(dir)) {
           addParentDirs(dir, zOut);
-          super.zipDir(null, zOut, dir + "/", ZipFileSet.DEFAULT_DIR_MODE, JAR_MARKER);
+          super.zipDir((File)null, zOut, dir + "/", ZipFileSet.DEFAULT_DIR_MODE, JAR_MARKER);
         }
       }
     }
