--- /usr/local/tmp/portage/dev-java/plexus-avalon-personality-0.13/work/plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/AvalonLifecycleHandler.java	2007-04-22 18:52:59.000000000 +0200
+++ AvalonLifecycleHandler.java	2007-04-23 02:18:59.033633279 +0200
@@ -13,7 +13,6 @@
     extends AbstractLifecycleHandler
 {
     public void initialize()
-        throws Exception
     {
     }
 }
