--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ComposePhase.java	2007-04-22 18:52:59.006719733 +0200
+++ ComposePhase.java	2007-04-23 20:18:16.576686333 +0200
@@ -12,14 +12,17 @@
         ComponentManager.class.getName();
 
     public void execute( Object object, org.codehaus.plexus.component.manager.ComponentManager manager )
-        throws Exception
     {
         if ( object instanceof Composable )
         {
+			try{
             PlexusContainer container = manager.getContainer();
 
+
             ComponentManager avalonComponentManager = (ComponentManager) container.lookup( COMPONENT_MANAGER_ROLE );
 
+
+
             if ( null == avalonComponentManager )
             {
                 final String message = "ComponentManager is null";
@@ -29,5 +32,9 @@
 
             ( (Composable) object ).compose( avalonComponentManager );
         }
+			catch (Exception e){
+				e.printStackTrace();
+			}
+		}
     }
 }
