--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ContextualizePhase.java	2007-04-22 18:52:58.922712039 +0200
+++ ContextualizePhase.java	2007-04-23 20:51:28.791150479 +0200
@@ -11,8 +11,9 @@
     extends AbstractPhase
 {
     public void execute( Object object, ComponentManager manager )
-        throws Exception
+		/*        throws Exception*/
     {
+		try{ 
         if ( object instanceof Contextualizable )
         {
             PlexusContainer container = manager.getContainer();
@@ -29,4 +31,9 @@
             ( (Contextualizable) object ).contextualize( context );
         }
     }
+		catch(org.apache.avalon.framework.context.ContextException e)
+		{
+			e.printStackTrace();
+		} 
+	}
 }
