--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/LogDisablePhase.java	2007-04-22 18:52:59.018720833 +0200
+++ LogDisablePhase.java	2007-04-23 21:07:44.752537420 +0200
@@ -14,8 +14,10 @@
     extends AbstractPhase
 {
     public void execute( Object object, ComponentManager manager )
-        throws Exception
+		/*throws Exception*/
+
     {
+		try{ 
         if ( object instanceof LogEnabled )
         {
             PlexusContainer container = manager.getContainer();
@@ -35,4 +38,8 @@
             }
         }
     }
+		catch ( org.codehaus.plexus.component.repository.exception.ComponentLookupException e){
+			e.printStackTrace();
+		} 
+	}
 }
