diff -Nur plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/AvalonComponentRepository.java plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/AvalonComponentRepository.java
--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/AvalonComponentRepository.java	2007-04-22 19:52:59.000000000 +0300
+++ plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/AvalonComponentRepository.java	2007-05-13 17:40:29.000000000 +0300
@@ -19,7 +19,8 @@
 	/**
 	 * @see org.codehaus.plexus.component.repository.ComponentRepository#initialize()
 	 */
-	public void initialize() throws Exception
+	/* seem not to throw ...        public void initialize() throws Exception */
+	public void initialize() throws  org.codehaus.plexus.component.repository.exception.ComponentRepositoryException
 	{
 		super.initialize();
 	}
diff -Nur plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/AvalonLifecycleHandler.java plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/AvalonLifecycleHandler.java
--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/AvalonLifecycleHandler.java	2007-04-22 19:52:59.000000000 +0300
+++ plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/AvalonLifecycleHandler.java	2007-05-13 17:41:28.000000000 +0300
@@ -13,7 +13,6 @@
     extends AbstractLifecycleHandler
 {
     public void initialize()
-        throws Exception
     {
     }
 }
diff -Nur plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ComposePhase.java plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ComposePhase.java
--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ComposePhase.java	2007-04-22 19:52:59.000000000 +0300
+++ plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ComposePhase.java	2007-05-13 17:41:58.000000000 +0300
@@ -12,14 +12,17 @@
         ComponentManager.class.getName();
 
     public void execute( Object object, org.codehaus.plexus.component.manager.ComponentManager manager )
-        throws Exception
     {
         if ( object instanceof Composable )
         {
+			try{
             PlexusContainer container = manager.getContainer();
 
+
             ComponentManager avalonComponentManager = (ComponentManager) container.lookup( COMPONENT_MANAGER_ROLE );
 
+
+
             if ( null == avalonComponentManager )
             {
                 final String message = "ComponentManager is null";
@@ -29,5 +32,9 @@
 
             ( (Composable) object ).compose( avalonComponentManager );
         }
+			catch (Exception e){
+				e.printStackTrace();
+			}
+		}
     }
 }
diff -Nur plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ConfigurePhase.java plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ConfigurePhase.java
--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ConfigurePhase.java	2007-04-22 19:52:58.000000000 +0300
+++ plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ConfigurePhase.java	2007-05-13 17:42:12.000000000 +0300
@@ -12,8 +12,8 @@
     extends AbstractPhase
 {
     public void execute( Object object, ComponentManager manager )
-        throws Exception
     {
+		try{
         if ( object instanceof Configurable )
         {
             PlexusConfiguration plexusConfiguration = manager.getComponentDescriptor().getConfiguration();
@@ -28,4 +28,8 @@
             ( (Configurable) object ).configure( configuration );
         }
     }
+		catch( org.apache.avalon.framework.configuration.ConfigurationException e) {
+			e.printStackTrace();
+		}
+	}
 }
diff -Nur plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ContextualizePhase.java plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ContextualizePhase.java
--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ContextualizePhase.java	2007-04-22 19:52:58.000000000 +0300
+++ plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ContextualizePhase.java	2007-05-13 17:42:20.000000000 +0300
@@ -11,8 +11,9 @@
     extends AbstractPhase
 {
     public void execute( Object object, ComponentManager manager )
-        throws Exception
+		/*        throws Exception*/
     {
+		try{ 
         if ( object instanceof Contextualizable )
         {
             PlexusContainer container = manager.getContainer();
@@ -29,4 +30,9 @@
             ( (Contextualizable) object ).contextualize( context );
         }
     }
+		catch(org.apache.avalon.framework.context.ContextException e)
+		{
+			e.printStackTrace();
+		} 
+	}
 }
diff -Nur plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/DisposePhase.java plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/DisposePhase.java
--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/DisposePhase.java	2007-04-22 19:52:59.000000000 +0300
+++ plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/DisposePhase.java	2007-05-13 17:42:29.000000000 +0300
@@ -8,11 +8,17 @@
     extends AbstractPhase
 {
     public void execute( Object object, ComponentManager manager )
-        throws Exception
+		/*        throws Exception*/
     {
+/*		try{*/
         if ( object instanceof Disposable )
         {
             ( (Disposable) object ).dispose();
         }
+/*		}
+		catch ( Exception e){
+			e.printStackTrace();
+		}*/
     }
+
 }
diff -Nur plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/InitializePhase.java plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/InitializePhase.java
--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/InitializePhase.java	2007-04-22 19:52:58.000000000 +0300
+++ plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/InitializePhase.java	2007-05-13 17:42:34.000000000 +0300
@@ -8,11 +8,16 @@
     extends AbstractPhase
 {
     public void execute( Object object, ComponentManager manager )
-        throws Exception
+		/*        throws Exception*/
     {
+		try{
         if ( object instanceof Initializable )
         {
             ( (Initializable) object ).initialize();
         }
     }
+		catch (Exception e){
+			e.printStackTrace();
+		}
+	}
 }
diff -Nur plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/LogDisablePhase.java plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/LogDisablePhase.java
--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/LogDisablePhase.java	2007-04-22 19:52:59.000000000 +0300
+++ plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/LogDisablePhase.java	2007-05-13 17:42:44.000000000 +0300
@@ -14,8 +14,10 @@
     extends AbstractPhase
 {
     public void execute( Object object, ComponentManager manager )
-        throws Exception
+		/*throws Exception*/
+
     {
+		try{ 
         if ( object instanceof LogEnabled )
         {
             PlexusContainer container = manager.getContainer();
@@ -35,4 +37,8 @@
             }
         }
     }
+		catch ( org.codehaus.plexus.component.repository.exception.ComponentLookupException e){
+			e.printStackTrace();
+		} 
+	}
 }
diff -Nur plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/LogEnablePhase.java plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/LogEnablePhase.java
--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/LogEnablePhase.java	2007-04-22 19:52:59.000000000 +0300
+++ plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/LogEnablePhase.java	2007-05-13 17:42:51.000000000 +0300
@@ -15,8 +15,9 @@
     extends AbstractPhase
 {
     public void execute( Object object, ComponentManager manager )
-        throws Exception
+		/*        throws Exception*/
     {
+		try{
         if ( object instanceof LogEnabled )
         {
             PlexusContainer container = manager.getContainer();
@@ -47,4 +48,8 @@
             ( (LogEnabled) object ).enableLogging( logger );
         }
     }
+		catch (Exception e){
+			e.printStackTrace();
+	}
+}
 }
diff -Nur plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ReconfigurePhase.java plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ReconfigurePhase.java
--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ReconfigurePhase.java	2007-04-22 19:52:59.000000000 +0300
+++ plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ReconfigurePhase.java	2007-05-13 17:43:09.000000000 +0300
@@ -10,8 +10,9 @@
     extends AbstractPhase
 {
     public void execute( Object object, ComponentManager manager )
-        throws Exception
+		/*        throws Exception*/
     {
+		try{
         Configuration configuration = new AvalonConfiguration( manager.getComponentDescriptor().getConfiguration() );
 
         if ( object instanceof Reconfigurable )
@@ -24,4 +25,8 @@
             ( (Reconfigurable) object ).reconfigure( configuration );
         }
     }
+		catch(Exception e){
+			e.printStackTrace();
+		}
+	}
 }
diff -Nur plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/RecontextualizePhase.java plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/RecontextualizePhase.java
--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/RecontextualizePhase.java	2007-04-22 19:52:59.000000000 +0300
+++ plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/RecontextualizePhase.java	2007-05-13 17:43:32.000000000 +0300
@@ -11,8 +11,9 @@
     extends AbstractPhase
 {
     public void execute( Object object, ComponentManager manager )
-        throws Exception
+		/*throws Exception*/
     {
+		try{
         if ( object instanceof Recontextualizable )
         {
             PlexusContainer container = manager.getContainer();
@@ -28,5 +29,6 @@
 
             ( (Recontextualizable) object ).recontextualize( context );
         }
+		}catch(Exception e){e.printStackTrace();}
     }
 }
diff -Nur plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ResumePhase.java plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ResumePhase.java
--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ResumePhase.java	2007-04-22 19:52:58.000000000 +0300
+++ plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ResumePhase.java	2007-05-13 17:43:37.000000000 +0300
@@ -8,11 +8,13 @@
     extends AbstractPhase
 {
     public void execute( Object object, ComponentManager manager )
-        throws Exception
+		/*      throws Exception*/
     {
+		try{
         if ( object instanceof Suspendable )
         {
             ( (Suspendable) object ).resume();
         }
+		}catch (Exception e){e.printStackTrace();}
     }
 }
diff -Nur plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ServicePhase.java plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ServicePhase.java
--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ServicePhase.java	2007-04-22 19:52:58.000000000 +0300
+++ plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/ServicePhase.java	2007-05-13 17:43:42.000000000 +0300
@@ -14,8 +14,9 @@
 
     public void execute( Object object, ComponentManager manager
                          )
-        throws Exception
+		/*        throws Exception*/
     {
+		try{
         if ( object instanceof Serviceable )
         {
             PlexusContainer container = manager.getContainer();
@@ -31,5 +32,7 @@
 
             ( (Serviceable) object ).service( serviceManager );
         }
+		}catch(org.codehaus.plexus.component.repository.exception.ComponentLookupException e){e.printStackTrace();}
+		catch(Exception e){e.printStackTrace();}
     }
 }
diff -Nur plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/StartPhase.java plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/StartPhase.java
--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/StartPhase.java	2007-04-22 19:52:59.000000000 +0300
+++ plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/StartPhase.java	2007-05-13 17:43:49.000000000 +0300
@@ -8,11 +8,13 @@
     extends AbstractPhase
 {
     public void execute( Object object, ComponentManager manager )
-        throws Exception
+		/*        throws Exception*/
     {
+		try{
         if ( object instanceof Startable )
         {
             ( (Startable) object ).start();
         }
+		}catch (Exception e){e.printStackTrace();}
     }
 }
diff -Nur plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/StopPhase.java plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/StopPhase.java
--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/StopPhase.java	2007-04-22 19:52:59.000000000 +0300
+++ plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/StopPhase.java	2007-05-13 17:43:55.000000000 +0300
@@ -8,11 +8,13 @@
     extends AbstractPhase
 {
     public void execute( Object object, ComponentManager manager )
-        throws Exception
+		/*throws Exception*/
     {
+		try{
         if ( object instanceof Startable )
         {
             ( (Startable) object ).stop();
         }
+		}catch(Exception e){e.printStackTrace();}
     }
 }
diff -Nur plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/SuspendPhase.java plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/SuspendPhase.java
--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/SuspendPhase.java	2007-04-22 19:52:58.000000000 +0300
+++ plexus-avalon-personality-0.13_patched/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/SuspendPhase.java	2007-05-13 17:44:00.000000000 +0300
@@ -8,11 +8,13 @@
     extends AbstractPhase
 {
     public void execute( Object object, ComponentManager manager )
-        throws Exception
+		/*       throws Exception*/
     {
+		try{
         if ( object instanceof Suspendable )
         {
             ( (Suspendable) object ).suspend();
         }
+		}catch(Exception e){e.printStackTrace();}
     }
 }
