--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/StopPhase.java	2007-04-22 18:52:59.034722297 +0200
+++ StopPhase.java	2007-04-23 21:26:25.987229742 +0200
@@ -8,11 +8,13 @@
     extends AbstractPhase
 {
     public void execute( Object object, ComponentManager manager )
-        throws Exception
+		/*throws Exception*/
     {
+		try{
         if ( object instanceof Startable )
         {
             ( (Startable) object ).stop();
         }
+		}catch(Exception e){e.printStackTrace();}
     }
 }
