--- plexus-avalon-personality-0.13/src/main/java/org/codehaus/plexus/personality/avalon/lifecycle/phase/SuspendPhase.java	2007-04-22 18:52:58.998719001 +0200
+++ SuspendPhase.java	2007-04-23 21:28:14.069128813 +0200
@@ -8,11 +8,13 @@
     extends AbstractPhase
 {
     public void execute( Object object, ComponentManager manager )
-        throws Exception
+		/*       throws Exception*/
     {
+		try{
         if ( object instanceof Suspendable )
         {
             ( (Suspendable) object ).suspend();
         }
+		}catch(Exception e){e.printStackTrace();}
     }
 }
