--- plugins/lib/Munin/Plugin/Pgsql.pm_orig	2017-10-30 03:31:45.000000000 +0300
+++ plugins/lib/Munin/Plugin/Pgsql.pm	2017-11-27 14:19:25.020161981 +0300
@@ -475,10 +475,10 @@
 
     return if (defined $self->{detected_version});
 
-    my $r = $self->runquery("SELECT version()");
+    my $r = $self->runquery("SHOW server_version");
     my $v = $r->[0]->[0];
     die "Unable to detect PostgreSQL version\n"
-        unless ($v =~ /^PostgreSQL (\d+)\.(\d+)(\.\d+|devel|beta\d+|rc\d+)\b/);
+        unless ($v =~ /^(\d+)\.(\d+).*\b/);
     $self->{detected_version} = "$1.$2";
 }
 

