diff -Nur weka-3-5-7/src/weka/gui/boundaryvisualizer/BoundaryPanel.java weka-3-5-7_patched/src/weka/gui/boundaryvisualizer/BoundaryPanel.java
--- weka-3-5-7/src/weka/gui/boundaryvisualizer/BoundaryPanel.java	2007-05-01 13:24:48.000000000 +0300
+++ weka-3-5-7_patched/src/weka/gui/boundaryvisualizer/BoundaryPanel.java	2008-04-13 10:17:44.000000000 +0300
@@ -51,9 +51,13 @@
 import javax.swing.JPanel;
 import javax.swing.ToolTipManager;
 
-import com.sun.image.codec.jpeg.JPEGCodec;
-import com.sun.image.codec.jpeg.JPEGEncodeParam;
-import com.sun.image.codec.jpeg.JPEGImageEncoder;
+import java.util.Locale;
+import javax.imageio.IIOImage;
+import javax.imageio.ImageIO;
+import javax.imageio.ImageWriteParam;
+import javax.imageio.ImageWriter;
+import javax.imageio.plugins.jpeg.JPEGImageWriteParam;
+import javax.imageio.stream.ImageOutputStream;
 
 /**
  * BoundaryPanel. A class to handle the plotting operations
@@ -991,17 +995,20 @@
       BufferedOutputStream out = 
         new BufferedOutputStream(new FileOutputStream(fileName));
       
-      JPEGImageEncoder encoder = JPEGCodec.createJPEGEncoder(out);
-
       BufferedImage bi = new BufferedImage(m_panelWidth, m_panelHeight,
                                            BufferedImage.TYPE_INT_RGB);
       Graphics2D gr2 = bi.createGraphics();
       gr2.drawImage(m_osi, 0, 0, m_panelWidth, m_panelHeight, null);
 
-      JPEGEncodeParam param = encoder.getDefaultJPEGEncodeParam(bi);
-      param.setQuality(1.0f, false);
-      encoder.setJPEGEncodeParam(param);
-      encoder.encode(bi);
+      ImageWriteParam iwparam = new JPEGImageWriteParam(Locale.getDefault());
+      iwparam.setCompressionMode(ImageWriteParam.MODE_EXPLICIT);
+      iwparam.setCompressionQuality(1.0f);
+      ImageWriter iw = (ImageWriter)ImageIO.getImageWritersByFormatName("jpg").next();
+      ImageOutputStream ios = ImageIO.createImageOutputStream(out);
+      iw.setOutput(ios);
+      iw.write(null, new IIOImage(bi, null, null), iwparam);
+      iw.dispose();
+      ios.close();
       out.flush();
       out.close();
     } catch (Exception ex) {
diff -Nur weka-3-5-7/src/weka/gui/visualize/JPEGWriter.java weka-3-5-7_patched/src/weka/gui/visualize/JPEGWriter.java
--- weka-3-5-7/src/weka/gui/visualize/JPEGWriter.java	2007-05-01 13:24:52.000000000 +0300
+++ weka-3-5-7_patched/src/weka/gui/visualize/JPEGWriter.java	2008-04-13 10:23:13.000000000 +0300
@@ -32,9 +32,13 @@
 
 import javax.swing.JComponent;
 
-import com.sun.image.codec.jpeg.JPEGCodec;
-import com.sun.image.codec.jpeg.JPEGEncodeParam;
-import com.sun.image.codec.jpeg.JPEGImageEncoder;
+import java.util.Locale;
+import javax.imageio.IIOImage;
+import javax.imageio.ImageIO;
+import javax.imageio.ImageWriteParam;
+import javax.imageio.ImageWriter;
+import javax.imageio.plugins.jpeg.JPEGImageWriteParam;
+import javax.imageio.stream.ImageOutputStream;
 
 /** 
  * This class takes any JComponent and outputs it to a JPEG-file.
@@ -156,8 +160,6 @@
    */
   public void generateOutput() throws Exception {
     BufferedImage		bi;
-    JPEGImageEncoder		encoder;
-    JPEGEncodeParam		param;
     Graphics			g;
     BufferedOutputStream	ostream;
 
@@ -170,11 +172,15 @@
       ((Graphics2D) g).scale(getXScale(), getYScale());
     g.fillRect(0, 0, getComponent().getWidth(), getComponent().getHeight());
     getComponent().paint(g);
-    encoder = JPEGCodec.createJPEGEncoder(ostream);
-    param   = encoder.getDefaultJPEGEncodeParam(bi);
-    param.setQuality(getQuality(), false);
-    encoder.setJPEGEncodeParam(param);
-    encoder.encode(bi);
+    ImageWriteParam iwparam = new JPEGImageWriteParam(Locale.getDefault());
+    iwparam.setCompressionMode(ImageWriteParam.MODE_EXPLICIT);
+    iwparam.setCompressionQuality(getQuality());
+    ImageWriter iw = (ImageWriter)ImageIO.getImageWritersByFormatName("jpg").next();
+    ImageOutputStream ios = ImageIO.createImageOutputStream(ostream);
+    iw.setOutput(ios);
+    iw.write(null, new IIOImage(bi, null, null), iwparam);
+    iw.dispose();
+    ios.close();
     ostream.flush();
     ostream.close();
   }
