diff -Naur work_orig/o/jdk/make/sun/splashscreen/FILES_c.gmk work/o/jdk/make/sun/splashscreen/FILES_c.gmk
--- work_orig/o/jdk/make/sun/splashscreen/FILES_c.gmk	2008-01-12 20:16:00.000000000 -0500
+++ work/o/jdk/make/sun/splashscreen/FILES_c.gmk	2008-01-12 20:17:31.000000000 -0500
@@ -30,8 +30,10 @@
 	splashscreen_impl.c \
 	splashscreen_jpeg.c \
 	splashscreen_png.c \
-	splashscreen_sys.c \
-	png.c \
+	splashscreen_sys.c
+
+ifneq ($(EXTERNAL_LIBPNG), true)
+FILES_c += png.c \
 	pngerror.c \
 	pngget.c \
 	pngmem.c  \
@@ -46,6 +48,7 @@
 	pngwrite.c \
 	pngwtran.c \
 	pngwutil.c
+endif
 
 ifneq ($(EXTERNAL_GIFLIB), true)
 FILES_c = dgif_lib.c \
diff -Naur work_orig/o/jdk/make/sun/splashscreen/Makefile work/o/jdk/make/sun/splashscreen/Makefile
--- work_orig/o/jdk/make/sun/splashscreen/Makefile	2008-01-12 20:16:00.000000000 -0500
+++ work/o/jdk/make/sun/splashscreen/Makefile	2008-01-12 20:17:59.000000000 -0500
@@ -82,7 +82,11 @@
   vpath %.c   $(SHARE_SRC)/native/$(PKGDIR)/giflib
 endif
 vpath %.c   $(SHARE_SRC)/native/java/util/zip/zlib-1.1.3
-vpath %.c   $(SHARE_SRC)/native/$(PKGDIR)/libpng
+ifeq ($(EXTERNAL_LIBPNG), true)
+  OTHER_LDLIBS += -lpng
+else
+  vpath %.c   $(SHARE_SRC)/native/$(PKGDIR)/libpng
+endif
 vpath %.c   $(SHARE_SRC)/native/$(PKGDIR)/image/jpeg
 vpath %.c   $(PLATFORM_SRC)/native/$(PKGDIR)/splashscreen
 
diff -Naur work_orig/o/jdk/src/share/native/sun/awt/splashscreen/splashscreen_png.c work/o/jdk/src/share/native/sun/awt/splashscreen/splashscreen_png.c
--- work_orig/o/jdk/src/share/native/sun/awt/splashscreen/splashscreen_png.c	2008-01-12 20:15:57.000000000 -0500
+++ work/o/jdk/src/share/native/sun/awt/splashscreen/splashscreen_png.c	2008-01-12 20:18:20.000000000 -0500
@@ -25,7 +25,7 @@
 
 #include "splashscreen_impl.h"
 
-#include "../libpng/png.h"
+#include "png.h"
 
 #include <setjmp.h>
 
