http://dev.gentooexperimental.org/~flameeyes/openjdk-patches/

Index: openjdk/j2se/make/sun/awt/mawt.gmk
===================================================================
--- openjdk/j2se/make/sun/awt/mawt.gmk.orig	2007-08-02 09:17:57.000000000 +0200
+++ openjdk/j2se/make/sun/awt/mawt.gmk	2007-08-02 23:45:28.000000000 +0200
@@ -58,7 +58,11 @@
     FILES_c += awt_motif21.c
     FILES_c += awt_Choice21.c
     ifeq  ($(PLATFORM), linux)
-        FILES_c += Xinerama.c
+       ifeq ($(EXTERNAL_LIBXINERAMA), true)
+         LIBXINERAMA = -lXinerama
+       else
+          FILES_c += Xinerama.c
+       endif
     endif
   endif
 
@@ -179,7 +183,7 @@
 OTHER_CFLAGS += -DMLIB_NO_LIBSUNMATH
 OTHER_CFLAGS += -DMOTIF_VERSION=2
 OTHER_CFLAGS += -DXMSTRINGDEFINES=1
-OTHER_LDLIBS = $(LIBXM) $(LIBXMU) $(LIBXTST) -lXext $(LIBXT) $(LIBSM) $(LIBICE) -lX11 -lXi
+OTHER_LDLIBS = $(LIBXM) $(LIBXMU) $(LIBXTST) $(LIBXINERAMA) -lXext $(LIBXT) $(LIBSM) $(LIBICE) -lX11 -lXi
 endif
 
 endif
Index: openjdk/j2se/make/sun/xawt/Makefile
===================================================================
--- openjdk.orig/j2se/make/sun/xawt/Makefile
+++ openjdk/j2se/make/sun/xawt/Makefile
@@ -53,7 +53,11 @@ AUTO_JAVA_PRUNE = WrapperGenerator.java
 LDFLAGS += -L$(OPENWIN_LIB)
 
 ifeq ($(PLATFORM), linux)
+ifeq ($(EXTERNAL_LIBXINERAMA), true)
+LDFLAGS += -lXinerama
+else
 FILES_c += Xinerama.c
+endif
 LDFLAGS += -lpthread
 dummy := $(shell $(MKDIR) -p $(LIB_LOCATION))
 endif
