Index: openjdk/j2se/make/sun/splashscreen/FILES_c.gmk
===================================================================
--- openjdk/j2se/make/sun/splashscreen/FILES_c.gmk.orig	2007-08-03 01:22:49.000000000 +0200
+++ openjdk/j2se/make/sun/splashscreen/FILES_c.gmk	2007-08-03 01:25:23.000000000 +0200
@@ -48,8 +48,10 @@
 	pngwutil.c \
 	dgif_lib.c \
 	gif_err.c \
-	gifalloc.c \
-	jcomapi.c \
+	gifalloc.c
+
+ifneq ($(EXTERNAL_JPEG), true)
+FILES_c += jcomapi.c \
 	jdapimin.c \
 	jdapistd.c \
 	jdcoefct.c \
@@ -93,6 +95,7 @@
 	jfdctflt.c \
 	jfdctfst.c \
 	jfdctint.c
+endif
 
 ifneq ($(EXTERNAL_ZLIB),true)
 FILES_c += compress.c \
Index: openjdk/j2se/make/sun/splashscreen/Makefile
===================================================================
--- openjdk.orig/j2se/make/sun/splashscreen/Makefile
+++ openjdk/j2se/make/sun/splashscreen/Makefile
@@ -83,13 +83,18 @@ vpath %.c   $(SHARE_SRC)/native/$(PKGDIR
 vpath %.c   $(PLATFORM_SRC)/native/$(PKGDIR)/splashscreen
 
 CPPFLAGS += -I$(PLATFORM_SRC)/native/$(PKGDIR)/splashscreen -I$(SHARE_SRC)/native/$(PKGDIR)/splashscreen
-CPPFLAGS += -I$(SHARE_SRC)/native/$(PKGDIR)/image/jpeg
 
 #
 # Has been converted to the JNI: generate JNI-style header files
 #
 JAVAHFLAGS += -jni
 
+ifeq ($(EXTERNAL_JPEG), true)
+  OTHER_LDLIBS += -ljpeg
+else
+  CPPFLAGS += -I$(SHARE_SRC)/native/$(PKGDIR)/image/jpeg
+endif
+
 ifeq ($(EXTERNAL_ZLIB), true)
   OTHER_LDLIBS += -lz
 else
Index: openjdk/j2se/src/share/native/sun/awt/splashscreen/splashscreen_jpeg.c
===================================================================
--- openjdk.orig/j2se/src/share/native/sun/awt/splashscreen/splashscreen_jpeg.c
+++ openjdk/j2se/src/share/native/sun/awt/splashscreen/splashscreen_jpeg.c
@@ -25,7 +25,6 @@
 
 #include "splashscreen_impl.h"
 
-#include "jinclude.h"
 #include "jpeglib.h"
 #include "jerror.h"
 
@@ -107,11 +106,11 @@ set_stream_src(j_decompress_ptr cinfo, S
     if (cinfo->src == NULL) {   /* first time for this JPEG object? */
         cinfo->src = (struct jpeg_source_mgr *)
             (*cinfo->mem->alloc_small) ((j_common_ptr) cinfo, 
-            JPOOL_PERMANENT, SIZEOF(stream_source_mgr));
+            JPOOL_PERMANENT, sizeof(stream_source_mgr));
         src = (stream_src_ptr) cinfo->src;
         src->buffer = (JOCTET *)
             (*cinfo->mem->alloc_small) ((j_common_ptr) cinfo, 
-            JPOOL_PERMANENT, INPUT_BUF_SIZE * SIZEOF(JOCTET));
+            JPOOL_PERMANENT, INPUT_BUF_SIZE * sizeof(JOCTET));
     }
 
     src = (stream_src_ptr) cinfo->src;
