Index: openjdk/j2se/make/sun/splashscreen/FILES_c.gmk
===================================================================
--- openjdk/j2se/make/sun/splashscreen/FILES_c.gmk.orig	2007-08-03 01:29:23.000000000 +0200
+++ openjdk/j2se/make/sun/splashscreen/FILES_c.gmk	2007-08-03 01:30:30.000000000 +0200
@@ -30,8 +30,10 @@
 	splashscreen_impl.c \
 	splashscreen_jpeg.c \
 	splashscreen_png.c \
-	splashscreen_sys.c \
-	png.c \
+	splashscreen_sys.c
+
+ifneq ($(EXTERNAL_LIBPNG), true)
+FILES_c += png.c \
 	pngerror.c \
 	pngget.c \
 	pngmem.c  \
@@ -46,6 +48,7 @@
 	pngwrite.c \
 	pngwtran.c \
 	pngwutil.c
+endif
 
 ifneq ($(EXTERNAL_GIFLIB), true)
 FILES_c = dgif_lib.c \
Index: openjdk/j2se/make/sun/splashscreen/Makefile
===================================================================
--- openjdk/j2se/make/sun/splashscreen/Makefile.orig	2007-08-02 09:17:58.000000000 +0200
+++ openjdk/j2se/make/sun/splashscreen/Makefile	2007-08-02 23:56:32.000000000 +0200
@@ -78,7 +78,11 @@
 vpath %.c   $(SHARE_SRC)/native/$(PKGDIR)
 vpath %.c   $(SHARE_SRC)/native/$(PKGDIR)/giflib
 vpath %.c   $(SHARE_SRC)/native/java/util/zip/zlib-1.1.3
-vpath %.c   $(SHARE_SRC)/native/$(PKGDIR)/libpng
+ifeq ($(EXTERNAL_LIBPNG), true)
+  OTHER_LDLIBS += -lpng
+else
+  vpath %.c   $(SHARE_SRC)/native/$(PKGDIR)/libpng
+endif
 vpath %.c   $(SHARE_SRC)/native/$(PKGDIR)/image/jpeg
 vpath %.c   $(PLATFORM_SRC)/native/$(PKGDIR)/splashscreen
 
Index: openjdk/j2se/src/share/native/sun/awt/splashscreen/splashscreen_png.c
===================================================================
--- openjdk.orig/j2se/src/share/native/sun/awt/splashscreen/splashscreen_png.c
+++ openjdk/j2se/src/share/native/sun/awt/splashscreen/splashscreen_png.c
@@ -25,7 +25,7 @@
 
 #include "splashscreen_impl.h"
 
-#include "../libpng/png.h"
+#include "png.h"
 
 #include <setjmp.h>
 
