Index: openjdk/j2se/make/sun/splashscreen/FILES_c.gmk
===================================================================
--- openjdk/j2se/make/sun/splashscreen/FILES_c.gmk.orig	2007-08-03 01:26:32.000000000 +0200
+++ openjdk/j2se/make/sun/splashscreen/FILES_c.gmk	2007-08-03 01:27:58.000000000 +0200
@@ -45,10 +45,13 @@
 	pngwio.c \
 	pngwrite.c \
 	pngwtran.c \
-	pngwutil.c \
-	dgif_lib.c \
+	pngwutil.c
+
+ifneq ($(EXTERNAL_GIFLIB), true)
+FILES_c = dgif_lib.c \
 	gif_err.c \
 	gifalloc.c
+endif
 
 ifneq ($(EXTERNAL_JPEG), true)
 FILES_c += jcomapi.c \
Index: openjdk/j2se/make/sun/splashscreen/Makefile
===================================================================
--- openjdk/j2se/make/sun/splashscreen/Makefile.orig	2007-08-02 09:17:58.000000000 +0200
+++ openjdk/j2se/make/sun/splashscreen/Makefile	2007-08-03 00:04:46.000000000 +0200
@@ -76,7 +76,11 @@
 #
 vpath %.c   $(SHARE_SRC)/native/$(PKGDIR)/splashscreen
 vpath %.c   $(SHARE_SRC)/native/$(PKGDIR)
-vpath %.c   $(SHARE_SRC)/native/$(PKGDIR)/giflib
+ifeq ($(EXTERNAL_GIFLIB), true)
+  OTHER_LDLIBS += -lgif
+else
+  vpath %.c   $(SHARE_SRC)/native/$(PKGDIR)/giflib
+endif
 vpath %.c   $(SHARE_SRC)/native/java/util/zip/zlib-1.1.3
 vpath %.c   $(SHARE_SRC)/native/$(PKGDIR)/libpng
 vpath %.c   $(SHARE_SRC)/native/$(PKGDIR)/image/jpeg
Index: openjdk/j2se/src/share/native/sun/awt/splashscreen/splashscreen_gif.c
===================================================================
--- openjdk.orig/j2se/src/share/native/sun/awt/splashscreen/splashscreen_gif.c
+++ openjdk/j2se/src/share/native/sun/awt/splashscreen/splashscreen_gif.c
@@ -26,7 +26,7 @@
 #include "splashscreen_impl.h"
 #include "splashscreen_gfx.h"
 
-#include "../giflib/gif_lib.h"
+#include "gif_lib.h"
 
 #define GIF_TRANSPARENT     0x01
 #define GIF_USER_INPUT      0x02
