Index: openjdk/hotspot/build/linux/makefiles/rules.make
===================================================================
--- openjdk.orig/hotspot/build/linux/makefiles/rules.make
+++ openjdk/hotspot/build/linux/makefiles/rules.make
@@ -37,8 +37,8 @@ DEMANGLER       = c++filt
 DEMANGLE        = $(DEMANGLER) < $@ > .$@ && mv -f .$@ $@
 
 # $(CC) is the c compiler (cc/gcc), $(CCC) is the c++ compiler (CC/g++).
-C_COMPILE       = $(CC) $(CPPFLAGS) $(CFLAGS)
-CC_COMPILE      = $(CCC) $(CPPFLAGS) $(CFLAGS)
+C_COMPILE       = $(CC) $(CPPFLAGS) $(CFLAGS) $(OTHER_CFLAGS)
+CC_COMPILE      = $(CCC) $(CPPFLAGS) $(CFLAGS) $(OTHER_CXXFLAGS)
 
 AS.S            = $(AS) $(ASFLAGS)
 
