Index: openjdk/j2se/make/com/sun/java/pack/Makefile
===================================================================
--- openjdk.orig/j2se/make/com/sun/java/pack/Makefile
+++ openjdk/j2se/make/com/sun/java/pack/Makefile
@@ -85,7 +85,7 @@ ifeq ($(STANDALONE),true)
   OTHER_CXXFLAGS += $(ZINCLUDE)
   LDDFLAGS += $(ZIPOBJS)
 else
-  OTHER_CXXFLAGS += -DNO_ZLIB -DUNPACK_JNI
+  CXXFLAGS_COMMON += -DNO_ZLIB -DUNPACK_JNI
   OTHER_LDLIBS += $(JVMLIB)
 endif
 
