Index: jaxb-ri-1.0.6/meta-inf/api/MANIFEST.MF
===================================================================
--- /dev/null
+++ jaxb-ri-1.0.6/meta-inf/api/MANIFEST.MF
@@ -0,0 +1,9 @@
+Manifest-Version: 1.0
+Specification-Title: Java Architecture for XML Binding
+Specification-Version: 1.0
+Specification-Vendor: Sun Microsystems, Inc.
+Implementation-Title: JAXB Reference Implementation 
+Implementation-Version: 1.0.1
+Implementation-Vendor: Sun Microsystems, Inc.
+Implementation-Vendor-Id: com.sun
+Extension-Name: javax.xml.bind
Index: jaxb-ri-1.0.6/meta-inf/impl/MANIFEST.MF
===================================================================
--- /dev/null
+++ jaxb-ri-1.0.6/meta-inf/impl/MANIFEST.MF
@@ -0,0 +1,9 @@
+Manifest-Version: 1.0
+Specification-Title: Java Architecture for XML Binding
+Specification-Version: 1.0
+Specification-Vendor: Sun Microsystems, Inc.
+Implementation-Title: JAXB Reference Implementation 
+Implementation-Version: 1.0.6
+Implementation-Vendor: Sun Microsystems, Inc.
+Implementation-Vendor-Id: com.sun
+Extension-Name: com.sun.xml.bind
Index: jaxb-ri-1.0.6/meta-inf/xjc/MANIFEST.MF
===================================================================
--- /dev/null
+++ jaxb-ri-1.0.6/meta-inf/xjc/MANIFEST.MF
@@ -0,0 +1,10 @@
+Manifest-Version: 1.0
+Specification-Title: Java Architecture for XML Binding
+Specification-Version: 1.0
+Specification-Vendor: Sun Microsystems, Inc.
+Implementation-Title: JAXB Reference Implementation 
+Implementation-Version: 1.0.6
+Implementation-Vendor: Sun Microsystems, Inc.
+Implementation-Vendor-Id: com.sun
+Extension-Name: com.sun.tools.xjc
+Main-Class: com.sun.tools.xjc.Driver
Index: jaxb-ri-1.0.6/meta-inf/xjc/services/com.sun.tools.xjc.CodeAugmenter
===================================================================
--- /dev/null
+++ jaxb-ri-1.0.6/meta-inf/xjc/services/com.sun.tools.xjc.CodeAugmenter
@@ -0,0 +1,3 @@
+com.sun.tools.xjc.addon.locator.SourceLocationAddOn
+com.sun.tools.xjc.addon.sync.SynchronizedMethodAddOn
+com.sun.tools.xjc.addon.elementwrapper.ElementWrapperAddOn
Index: jaxb-ri-1.0.6/meta-inf/xjc/services/org.relaxng.datatype.DatatypeLibraryFactory
===================================================================
--- /dev/null
+++ jaxb-ri-1.0.6/meta-inf/xjc/services/org.relaxng.datatype.DatatypeLibraryFactory
@@ -0,0 +1 @@
+com.sun.relaxng.javadt.DatatypeLibraryImpl
