diff --git a/pinba.cc b/pinba.cc
index 5be9f6e..5ea204f 100644
--- a/pinba.cc
+++ b/pinba.cc
@@ -251,7 +251,7 @@ static int sapi_ub_write_counter(const char *str, unsigned int length TSRMLS_DC)
 }
 /* }}} */
 
-static int php_pinba_init_socket (void) /* {{{ */
+static int php_pinba_init_socket (TSRMLS_D = NULL) /* {{{ */
 {
 	struct addrinfo *ai_list;
 	struct addrinfo *ai_ptr;
@@ -259,6 +259,7 @@ static int php_pinba_init_socket (void) /* {{{ */
 	int fd;
 	int status;
 
+	if (TSRMLS_C == NULL) TSRMLS_FETCH();
 	if (PINBA_G(server_host) == NULL || PINBA_G(server_port) == NULL) {
 		return -1;
 	}
@@ -503,7 +504,7 @@ static void php_pinba_flush_data(const char *custom_script_name TSRMLS_DC) /* {{
 		return;
 	}
 
-	status = php_pinba_init_socket();
+	status = php_pinba_init_socket(TSRMLS_C);
 	if (status != 0) {
 		PINBA_G(timers_stopped) = 0;
 		return;
