diff -Nur Jubler-3.9.0/src/com/panayotis/jubler/os/SystemFileFinder.java Jubler-3.9.0_patched/src/com/panayotis/jubler/os/SystemFileFinder.java
--- Jubler-3.9.0/src/com/panayotis/jubler/os/SystemFileFinder.java	2008-03-30 23:04:04.000000000 +0300
+++ Jubler-3.9.0_patched/src/com/panayotis/jubler/os/SystemFileFinder.java	2008-05-24 11:58:56.000000000 +0300
@@ -58,15 +58,12 @@
     }
     
     public static boolean loadLibrary(String name) {
-        File libfile = findFile("lib"+pathseparator+System.mapLibraryName(name));
-        if (libfile!=null) {
             try {
-                System.load(libfile.getAbsolutePath());
+                System.loadLibrary(name);
                 return true;
             } catch (UnsatisfiedLinkError e) {
                 e.printStackTrace();
             }
-        }
         return false;
     }
     
