diff -Nur Jubler-3.2.0/src/com/panayotis/jubler/Jubler.java Jubler-3.2.0_splitfix/src/com/panayotis/jubler/Jubler.java
--- Jubler-3.2.0/src/com/panayotis/jubler/Jubler.java	2007-02-27 23:39:26.000000000 +0200
+++ Jubler-3.2.0_splitfix/src/com/panayotis/jubler/Jubler.java	2007-03-11 11:32:14.000000000 +0200
@@ -1827,14 +1827,14 @@
                     subs2.add(csub);
                 }
             }
-            
+            Subtitles oldSubs = subs;
             setSubs(subs1);
             
             Jubler newwindow = new Jubler(subs2);
             newwindow.undo.invalidateSaveMark();
             
-            newwindow.setFile(new File(subs.getCurrentFile()+"_2"), true);
-            setFile(new File(subs.getCurrentFile()+"_1"), false);
+            newwindow.setFile(new File(oldSubs.getCurrentFile()+"_2"), true);
+            setFile(new File(oldSubs.getCurrentFile()+"_1"), false);
         }
     }//GEN-LAST:event_SplitTMActionPerformed
     
