--- maven-core-2.0.6/src/main/java/org/apache/maven/cli/MavenCli.java	2007-04-25 15:11:49.827180601 +0200
+++ MavenCli.java	2007-05-01 14:56:11.194727907 +0200
@@ -148,8 +148,9 @@
         // Now that we have everything that we need we will fire up plexus and
         // bring the maven component to life for use.
         // ----------------------------------------------------------------------
-
+        try{
         embedder = new Embedder();
+	}catch ( Exception e) { e.printStackTrace();}
 
         try
         {
