Index: maven-assembly-plugin-2.1_pre20070602/src/main/java/org/apache/maven/plugin/assembly/archive/archiver/PrefixingProxyArchiver.java
===================================================================
--- maven-assembly-plugin-2.1_pre20070602/src/main/java/org/apache/maven/plugin/assembly/archive/archiver/PrefixingProxyArchiver.java	(révision 543877)
+++ maven-assembly-plugin-2.1_pre20070602/src/main/java/org/apache/maven/plugin/assembly/archive/archiver/PrefixingProxyArchiver.java	(copie de travail)
@@ -7,6 +7,13 @@
 import java.io.IOException;
 import java.util.Map;
 
+/* correct this :
+ *     [javac] Compiling 67 source files to /usr/local/tmp/portage/dev-java/maven-assembly-plugin-2.1_pre20070602/work/maven-assembly-plugin-2.1_pre20070602/target/classes
+ *         [javac] /usr/local/tmp/portage/dev-java/maven-assembly-plugin-2.1_pre20070602/work/maven-assembly-plugin-2.1_pre20070602/src/main/java/org/apache/maven/plugin/assembly/archive/archiver/PrefixingProxyArchiver.java:10: org.apache.maven.plugin.assembly.archive.archiver.PrefixingProxyArchiver is not abstract and does not override abstract method addArchivedFileSet(org.codehaus.plexus.archiver.ArchivedFileSet) in org.codehaus.plexus.archiver.Archiver
+ *  
+ *      [javac] /usr/local/tmp/portage/dev-java/maven-assembly-plugin-2.1_pre20070602/work/maven-assembly-plugin-2.1_pre20070602/src/main/java/org/apache/maven/plugin/assembly/archive/archiver/PrefixingProxyArchiver.java:15: org.apache.maven.plugin.assembly.archive.archiver.PrefixingProxyArchiver is not abstract and does not override abstract method addFileSet(org.codehaus.plexus.archiver.FileSet) in org.codehaus.plexus.archiver.Archiver
+ *
+ */
 public class PrefixingProxyArchiver
     implements Archiver
 {
@@ -25,6 +32,18 @@
         }
     }
 
+     public void addArchivedFileSet( org.codehaus.plexus.archiver.ArchivedFileSet myset )
+        throws ArchiverException
+    {
+        delegate.addArchivedFileSet( myset);
+    }             
+
+      public void addFileSet( org.codehaus.plexus.archiver.FileSet myset )
+        throws ArchiverException
+    {
+        delegate.addFileSet( myset);
+    }              
+
     public void addArchivedFileSet( File archiveFile, String prefix, String[] includes, String[] excludes )
         throws ArchiverException
     {
