diff --git a/config.m4 b/config.m4
index 2cc15d6..23ed2d9 100644
--- a/config.m4
+++ b/config.m4
@@ -44,17 +44,17 @@ if test "$PHP_IGBINARY" != "no"; then
   if test ! -z "`$CC --version | grep -i GCC`"; then
     AC_MSG_RESULT(gcc)
     if test -z "`echo $CFLAGS | grep -- -O0`"; then
-      PHP_IGBINARY_CFLAGS="-Wall -Wpointer-arith -Wmissing-prototypes -Wstrict-prototypes -Wcast-align -Wshadow -Wwrite-strings -Wswitch -Winline -finline-limit=10000 --param large-function-growth=10000 --param inline-unit-growth=10000"
+      PHP_IGBINARY_CFLAGS="$CFLAGS -Wall -Wpointer-arith -Wmissing-prototypes -Wstrict-prototypes -Wcast-align -Wshadow -Wwrite-strings -Wswitch -Winline -finline-limit=10000 --param large-function-growth=10000 --param inline-unit-growth=10000"
     fi
   elif test ! -z "`$CC --version | grep -i ICC`"; then
     AC_MSG_RESULT(icc)
     if test -z "`echo $CFLAGS | grep -- -O0`"; then
-      PHP_IGBINARY_CFLAGS="-no-prec-div -O3 -x0 -unroll2"
+      PHP_IGBINARY_CFLAGS="$CFLAGS -no-prec-div -O3 -x0 -unroll2"
     fi
   elif test ! -z "`$CC --version | grep -i CLANG`"; then
     AC_MSG_RESULT(clang)
     if test -z "`echo $CFLAGS | grep -- -O0`"; then
-      PHP_IGBINARY_CFLAGS="-Wall -O2"
+      PHP_IGBINARY_CFLAGS="$CFLAGS -Wall -O2"
     fi
   else
     AC_MSG_RESULT(other)
