--- a/Makefile.orig	2017-12-27 15:28:56.067321595 +0300
+++ b/Makefile	2017-12-27 15:29:18.879322023 +0300
@@ -3451,9 +3451,9 @@
 IVYKIS_CFLAGS = 
 IVYKIS_LIBS = -livykis 
 IVYKIS_SUBDIRS = 
-JAR = /usr/libexec/eselect-java/run-java-tool.bash
-JAVAC = /usr/libexec/eselect-java/run-java-tool.bash
-JAVAH = /usr/libexec/eselect-java/run-java-tool.bash
+JAR = /usr/bin/jar
+JAVAC = /usr/bin/javac
+JAVAH = /usr/bin/javah
 JNI_CFLAGS = -I -I
 JNI_LIBS = -L -ljvm
 JSON_CFLAGS = -I/usr/include/json-c 
