--- a/Makefile.orig	2017-11-22 16:04:22.193550393 +0300
+++ b/Makefile	2017-11-22 16:07:28.413553889 +0300
@@ -3171,9 +3171,9 @@
 IVYKIS_CFLAGS = -I$(top_srcdir)/lib/ivykis/src/include -I$(top_builddir)/lib/ivykis/src/include
 IVYKIS_LIBS = -Wl,--whole-archive -L$(top_builddir)/lib/ivykis/src -livykis -Wl,--no-whole-archive
 IVYKIS_SUBDIRS = lib/ivykis
-JAR = /usr/libexec/eselect-java/run-java-tool.bash
-JAVAC = /usr/libexec/eselect-java/run-java-tool.bash
-JAVAH = /usr/libexec/eselect-java/run-java-tool.bash
+JAR = /usr/bin/jar
+JAVAC = /usr/bin/javac
+JAVAH = /usr/bin/javah
 JNI_CFLAGS = -I -I
 JNI_LIBS = -L -ljvm
 JSON_CFLAGS = 

