--- Foundation/include/Poco/RegularExpression.h.orig	2015-08-04 08:20:55.000000000 +0300
+++ Foundation/include/Poco/RegularExpression.h	2015-12-11 16:55:36.559741499 +0300
@@ -26,17 +26,12 @@
 #include "Poco/Foundation.h"
 #include <vector>
 
-
-//
-// Copy these definitions from pcre.h
-// to avoid pulling in the entire header file
-//
-extern "C"
-{
-	struct real_pcre8_or_16;                 /* declaration; the definition is private  */
-	typedef struct real_pcre8_or_16 pcre;
-	struct pcre_extra;
-}
+#if defined(POCO_UNBUNDLED)
+#include <pcre.h>
+#else
+#include "pcre_config.h"
+#include "pcre.h"
+#endif
 
 
 namespace Poco {
