--- Foundation/src/RegularExpression.cpp.orig	2015-08-04 08:20:56.000000000 +0300
+++ Foundation/src/RegularExpression.cpp	2015-12-11 16:49:11.139741545 +0300
@@ -14,15 +14,9 @@
 //
 
 
-#include "Poco/RegularExpression.h"
-#include "Poco/Exception.h"
 #include <sstream>
-#if defined(POCO_UNBUNDLED)
-#include <pcre.h>
-#else
-#include "pcre_config.h"
-#include "pcre.h"
-#endif
+#include "Poco/Exception.h"
+#include "Poco/RegularExpression.h"
 
 
 namespace Poco {
