--- src/java/com/opensymphony/module/sitemesh/html/tokenizer/lexer.flex.orig	2005-08-10 17:51:34.000000000 +0200
+++ src/java/com/opensymphony/module/sitemesh/html/tokenizer/lexer.flex	2007-07-23 11:38:49.000000000 +0200
@@ -29,7 +29,7 @@
 %{
     // Additional methods to add to generated Lexer to aid in error reporting.
     protected int position() { return yychar; }
-    protected int length()   { return yy_markedPos - yy_startRead; }
+    protected int length()   { return zzMarkedPos - zzStartRead; }
     protected int line()     { return -1; /*yyline;*/ }   // useful for debugging, but adds overhead
     protected int column()   { return -1; /*yycolumn;*/ } // useful for debugging, but adds overhead
     protected void resetLexerState() { yybegin(YYINITIAL); }
