--- src/org/apache/struts/util/SecureRequestUtils.java	2004-09-26 16:02:24.000000000 +0200
+++ src/org/apache/struts/util/SecureRequestUtils.java~	2009-03-24 20:47:53.000000000 +0100
@@ -358,10 +358,10 @@
          return false;
       }
 
-      Enumeration enum = aRequest.getAttributeNames();
+      Enumeration enu = aRequest.getAttributeNames();
       Map map = new HashMap();
-      while (enum.hasMoreElements()) {
-         String name = (String) enum.nextElement();
+      while (enu.hasMoreElements()) {
+         String name = (String) enu.nextElement();
          map.put(name, aRequest.getAttribute(name));
       }
       aRequest.getSession().setAttribute(STOWED_REQUEST_ATTRIBS, map);
@@ -534,9 +534,9 @@
     */
    private static Map getParameterMap(HttpServletRequest request) {
       Map map = new HashMap();
-      Enumeration enum = request.getParameterNames();
-      while (enum.hasMoreElements()) {
-         String name = (String) enum.nextElement();
+      Enumeration enu = request.getParameterNames();
+      while (enu.hasMoreElements()) {
+         String name = (String) enu.nextElement();
          String[] values = request.getParameterValues(name);
          map.put(name, values);
       }
