--- library/eclipse.c.orig	2007-09-08 22:17:52.000000000 +1200
+++ library/eclipse.c	2007-09-08 22:23:05.000000000 +1200
@@ -374,6 +374,7 @@
 	        _stprintf( errorMsg, homeMsg, officialName );
 	        displayMessage( officialName, errorMsg );
 	        free( errorMsg );
+		errorMsg = NULL;
     	}
     	exit( 1 );
     }
@@ -388,6 +389,7 @@
 	    	_stprintf( errorMsg, noVMMsg, officialName, msg );
 	    	displayMessage( officialName, errorMsg );
 	    	free( errorMsg );
+		errorMsg = NULL;
     	}
     	free( msg );
     	exit(1);
@@ -401,6 +403,7 @@
 	        _stprintf( errorMsg, startupMsg, officialName );
 	        displayMessage( officialName, errorMsg );
 	        free( errorMsg );
+		errorMsg = NULL;
 		}
     	exit( 1 );
 	}
@@ -529,6 +532,7 @@
 		            }
 		            displayMessage( title, errorMsg );
 		            free( errorMsg );
+			    errorMsg = NULL;
 	            }
 	            free( title );
 	            break;
