--- ./configure.ac.orig	2008-03-28 10:20:00.000000000 +0000
+++ ./configure.ac	2008-03-28 10:21:40.000000000 +0000
@@ -244,8 +244,8 @@
 
 dnl Check for plugin support headers and libraries.
 if test "$ENABLE_PLUGIN" = "yes"; then
-PKG_CHECK_MODULES(MOZILLA, mozilla-plugin, [MOZILLA_FOUND=yes], \
-  [MOZILLA_FOUND=no])
+PKG_CHECK_MODULES(MOZILLA, xulrunner-plugin xulrunner-xpcom, \
+  [MOZILLA_FOUND=yes], [MOZILLA_FOUND=no])
 if test "x${MOZILLA_FOUND}" = xno
 then
   PKG_CHECK_MODULES(MOZILLA, firefox-plugin firefox-xpcom, \
@@ -253,11 +253,6 @@
 fi
 if test "x${MOZILLA_FOUND}" = xno
 then
-  PKG_CHECK_MODULES(MOZILLA, xulrunner-plugin xulrunner-xpcom, \
-    [MOZILLA_FOUND=yes], [MOZILLA_FOUND=no])
-fi
-if test "x${MOZILLA_FOUND}" = xno
-then
   PKG_CHECK_MODULES(MOZILLA, mozilla-firefox-plugin mozilla-firefox-xpcom, \
     [MOZILLA_FOUND=yes], [MOZILLA_FOUND=no])
 fi
